lib/pos_blockchain.go

```diff
@@ -1562,8 +1566,6 @@ func (bc *Blockchain) removeTipBlockFromBestChain() *BlockNode {
 	lastBlock := bc.bestChain[len(bc.bestChain)-1]
 	delete(bc.bestChainMap, *lastBlock.Hash)
 	bc.bestChain = bc.bestChain[:len(bc.bestChain)-1]
-	bc.bestHeaderChain = bc.bestHeaderChain[:len(bc.bestChain)]
```

Comment:
Every time the bestChain mutated, this had the side effect of mutating the best header chain as well. This is not desirable during syncing where the blocksync process first syncs the header chain to the network's tip, and then syncs the block chain to the network's tip.